### PR TITLE
design notification model

### PR DESCRIPTION
**Pull request by TDM1911**
## Related Tickets:
- [#31883](https://edu-redmine.sun-asterisk.vn/issues/31883)

## WHAT this PR does?
- Add Notify Event.
- Add ProjectApproved Notification.
- Add notification functions for UserRepository.

## Notes Impacted Areas:
- N/A.

## Attachments:
- N/A.

### CODE REVIEW

**Review by kienpham9819 — APPROVED** [review]

**Comment by TDM1911** [comment]
ready

**Comment by TDM1911** [comment]
ready

File: app/Notifications/ProjectApproved.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+     * @param  mixed  $notifiable
+     * @return \Illuminate\Notifications\Messages\MailMessage
+     */
+    public function toMail($notifiable)
+    {
+        return (new MailMessage)
+                    ->line('The introduction to the notification.')
+                    ->action('Notification Action', url('/'))
+                    ->line('Thank you for using our application!');
+    }
```
Comment (huyenttm-0901) [inline app/Notifications/ProjectApproved.php]: hàm này e có dùng ko?
indent, i18n, ko dùng url

File: app/Repositories/User/UserRepositoryInterface.php [hunk 2]
```diff
@@ -43,4 +43,8 @@ public function hasPermissionTo($user, $permission);
     public function storeGithubToken($id, $token);
 
     public function getGithubRepositories($id, $userAgent);
+
+    public function getNotifications($id);
+
+    public function markAsRead($user, $notification);
```
Comment (huyenttm-0901) [inline app/Repositories/User/UserRepositoryInterface.php]:
```suggestion
    public function markAsRead($user, $notificationId);
```

File: app/Notifications/ProjectApproved.php (hunk 1 above)
Comment (TDM1911) [reply app/Notifications/ProjectApproved.php]: hàm tự render ra em quên chưa xóa ạ

**Comment by TDM1911** [comment]
ready

**Pull request merged**